csrc/transform_replay.cpp

```diff
@@ -200,6 +200,7 @@ TensorDomain* TransformReplay::fullSelfReplay(
   {
     size_t i = 0;
     for (auto id : self->root()) {
+#if 0
```

Comment:
I think what fails here is because some of the domains in `self` is marked as an rfactor product, but not in the `new_self_root`. IIRC, root domains transformed by a reshape op are marked as rfactor product.

Reply:
That's exactly right! What's the right fix?

Reply:
A bit more manual, but using `IterDomainBuilder` directly should work. You just don't want to call `resetRFactor()`.

Reply:
To be clear, I'm referring to the code about cloning iter domains.

Reply:
The source of the cloning, `in->getMaybeRFactorDomain()`, has no rfactor products. So changing cloning doesn't help. It's `r0->domain()`, the domain to replay, that has rfactor products.

Reply:
Yeah, but we know which domain should be rfactor products, so we could do `IterDomainBuilder::is_rfactor_domain(true)` for those domains.

Reply:
Yes, we could, but is it safe? 

In this particular case, the slice's output TV (and thus the original reshape's input) has an `rf` because `slice` explicitly adds it: https://github.com/NVIDIA/Fuser/blob/71d9b555b401655c0d664b94e8abb7021ce29637/csrc/ops/alias.cpp#L752. When we create the new reshape on the non-sliced input, the root domain of the new reshape shouldn't have the `rf` because those dimensions are not sliced in the first place.

(A bigger question from me is what the `rf` flag means...)

Reply:
Move the bigger, `rf` flag question to https://github.com/NVIDIA/Fuser/issues/1857.